torch/csrc/autograd/python_variable.cpp

```diff
@@ -240,6 +240,22 @@ class PyInterpreterHolder {
 };
 PyInterpreterHolder self_interpreter;
 
+c10::TensorImpl::SizesStridesPolicy parseSizesStridesPolicyArgument(
+    c10::string_view arg) {
+  if (arg == "strides") {
+    return c10::TensorImpl::SizesStridesPolicy::CustomStrides;
+  }
+
+  if (arg == "sizes") {
+    return c10::TensorImpl::SizesStridesPolicy::CustomSizes;
+  }
+
+  TORCH_CHECK_VALUE(
+      false,
+      "Unknown sizes_strides_policy: ",
+      arg,
+      "; expected 'strides' or 'sizes'");
```

Comment:
Is there some doc about what each values does?

Reply:
I can add documentation for it, where is are the docs for `_make_wrapper_subclass`/`_make_subclass` today?

As for the mismatch—I don't think that's fixable within the scope of this PR (and it already exists). But hopefully we'll finish coverage of the various tensorimpl methods soon and so at least we can document it?

Reply:
hmmmm, I feel like it's weird to put the rules for what this argument does in the error message; we should probably just have like a real docstring? 

Okay I'm going to merge this now, but I will take as a follow up:
- Finding the right place to describe the rules
- Making the code actually reflect the rules (e.g. make more things overridable from torchdispatch)